components/WalletConnectButton.jsx

```diff
@@ -540,22 +559,20 @@ const WalletConnectButton = props => {
 	};
 
 	const handleClickOutside = e => {
-		const dropDownNode = dropDownBox.current;
-		if (
-			buttonStyle &&
-			dropDownNode &&
-			!dropDownNode
-				.getAttribute('style')
-				.replace(/ /g, '')
-				.includes('display:none')
-		) {
+		const metamaskBoxNode = metamaskBoxRef.current;
+		const metamaskBoxOutsideNode = metamaskBoxOutsideRef.current;
+		if(buttonStyle
+		&& metamaskBoxNode
+		&& metamaskBoxOutsideNode) {
+			if(metamaskBoxNode.contains(e.target)) {
+				closedFromOutside.current = true;
+			}
 			// click outside the dropdown
-			// should have the same effect as toggling the button
-			closedFromOutside.current = true;
-			setTimeout(() => {
```

Comment:
i don't remember why I added the 300ms timeout, but there was a reason...
not sure what happens if we remove this, it could lead to weird closing situations.
No need to revert, but we need to test well if closing goes well under all circumstances

Reply:
Before, you could make general clicks outside of the MetaMask Box, even if the Metamask was closed. I have optimized it in such a way you can be clickable only if the MetaMask is open.

Maybe the **setTimeout** was there to prevent multiple clicks at the same time. 

But definitely, we need to test it.